### PR TITLE
Minor clean up of documents

### PR DESCRIPTION
**Pull request by elevran**
Small fixes for typos and markdown lint warnings (from vscode)

### CODE REVIEW

File: DEV-GUIDE.md [hunk 1]
```diff
@@ -146,14 +146,14 @@ export KUBECONFIG=${HOME}/.kube/config
...
+Execute once, to create a directory for the Istio source trees.
 
 ```shell
 mkdir -p $ISTIO
 ```
 
-As the steps recommmended in this section change both the user's groups
-information as well as the $PATH and environment, please logout of the
+As the steps recommended in this section change both the user's groups
+information as well as the $PATH and environment, please log out of the
```
Comment (ldemailly) [inline DEV-GUIDE.md]: let’s just delete that paragraph, there is no group change and no need to logout

Comment (elevran) [reply DEV-GUIDE.md]: Correct - updating to remove reference to groups. There are still changes to environment and PATH which require a log in.

Comment (ldemailly) [reply DEV-GUIDE.md]: no it requires to source or set the path, I don't think we should ask people to logout to change the path it sounds amateurish

File: DEV-GUIDE.md [hunk 2]
```diff
@@ -146,15 +146,14 @@ export KUBECONFIG=${HOME}/.kube/config
...
 
 ```shell
 mkdir -p $ISTIO
 ```
 
-As the steps recommmended in this section change both the user's groups
-information as well as the $PATH and environment, please logout of the
-development machine and log in to reload the environment.
+As the steps recommended in this section change the $PATH and environment,
+you will need to reload the environment.
```
Comment (ldemailly) [inline DEV-GUIDE.md]: good ! thanks !

**Pull request merged**